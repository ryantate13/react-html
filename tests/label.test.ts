import {label} from '../src';

describe('label', () => {
    it('renders with children', () => {
        expect(label({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(label({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(label()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(label({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(label({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(label({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(label({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(label({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(label({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(label({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(label({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(label({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(label({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(label({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(label({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(label({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(label({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(label({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(label({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(label({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(label({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(label({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(label({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(label({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(label({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(label({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(label({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(label({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(label({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(label({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(label({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(label({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(label({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(label({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(label({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(label({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(label({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(label({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(label({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(label({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(label({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(label({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(label({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(label({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(label({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(label({translate: "no"})).toMatchSnapshot();
    });

    it('renders with htmlFor="TEST"', () => {
        expect(label({htmlFor: "TEST"})).toMatchSnapshot();
    });
});