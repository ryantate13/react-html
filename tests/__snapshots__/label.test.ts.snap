// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label renders arrays of children 1`] = `
<label>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</label>
`;

exports[`label renders with accessKey="a,b,c,d" 1`] = `
<label
  accessKey="a,b,c,d"
/>
`;

exports[`label renders with autoCapitalize="characters" 1`] = `
<label
  autoCapitalize="characters"
/>
`;

exports[`label renders with autoCapitalize="none" 1`] = `
<label
  autoCapitalize="none"
/>
`;

exports[`label renders with autoCapitalize="off" 1`] = `
<label
  autoCapitalize="off"
/>
`;

exports[`label renders with autoCapitalize="on" 1`] = `
<label
  autoCapitalize="on"
/>
`;

exports[`label renders with autoCapitalize="sentences" 1`] = `
<label
  autoCapitalize="sentences"
/>
`;

exports[`label renders with autoCapitalize="words" 1`] = `
<label
  autoCapitalize="words"
/>
`;

exports[`label renders with children 1`] = `
<label>
  TEST
</label>
`;

exports[`label renders with className="foo bar baz" 1`] = `
<label
  className="foo bar baz"
/>
`;

exports[`label renders with contentEditable="false" 1`] = `
<label
  contentEditable="false"
/>
`;

exports[`label renders with contentEditable="true" 1`] = `
<label
  contentEditable="true"
/>
`;

exports[`label renders with dir="auto" 1`] = `
<label
  dir="auto"
/>
`;

exports[`label renders with dir="ltr" 1`] = `
<label
  dir="ltr"
/>
`;

exports[`label renders with dir="rtl" 1`] = `
<label
  dir="rtl"
/>
`;

exports[`label renders with draggable="false" 1`] = `
<label
  draggable="false"
/>
`;

exports[`label renders with draggable="true" 1`] = `
<label
  draggable="true"
/>
`;

exports[`label renders with empty attributes 1`] = `<label />`;

exports[`label renders with hidden=false 1`] = `
<label
  hidden={false}
/>
`;

exports[`label renders with hidden=true 1`] = `
<label
  hidden={true}
/>
`;

exports[`label renders with htmlFor="TEST" 1`] = `
<label
  htmlFor="TEST"
/>
`;

exports[`label renders with id="TEST" 1`] = `
<label
  id="TEST"
/>
`;

exports[`label renders with inputMode="decimal" 1`] = `
<label
  inputMode="decimal"
/>
`;

exports[`label renders with inputMode="email" 1`] = `
<label
  inputMode="email"
/>
`;

exports[`label renders with inputMode="none" 1`] = `
<label
  inputMode="none"
/>
`;

exports[`label renders with inputMode="numeric" 1`] = `
<label
  inputMode="numeric"
/>
`;

exports[`label renders with inputMode="search" 1`] = `
<label
  inputMode="search"
/>
`;

exports[`label renders with inputMode="tel" 1`] = `
<label
  inputMode="tel"
/>
`;

exports[`label renders with inputMode="text" 1`] = `
<label
  inputMode="text"
/>
`;

exports[`label renders with inputMode="url" 1`] = `
<label
  inputMode="url"
/>
`;

exports[`label renders with is="foo" 1`] = `
<label
  is="foo"
/>
`;

exports[`label renders with itemID="https://google.com/" 1`] = `
<label
  itemID="https://google.com/"
/>
`;

exports[`label renders with itemProp="https://google.com/" 1`] = `
<label
  itemProp="https://google.com/"
/>
`;

exports[`label renders with itemRef="foo bar" 1`] = `
<label
  itemRef="foo bar"
/>
`;

exports[`label renders with itemScope=false 1`] = `
<label
  itemScope={false}
/>
`;

exports[`label renders with itemScope=true 1`] = `
<label
  itemScope={true}
/>
`;

exports[`label renders with itemType="https://google.com/" 1`] = `
<label
  itemType="https://google.com/"
/>
`;

exports[`label renders with lang="" 1`] = `
<label
  lang=""
/>
`;

exports[`label renders with lang="es-419" 1`] = `
<label
  lang="es-419"
/>
`;

exports[`label renders with slot="TEST" 1`] = `
<label
  slot="TEST"
/>
`;

exports[`label renders with spellCheck="false" 1`] = `
<label
  spellCheck="false"
/>
`;

exports[`label renders with spellCheck="true" 1`] = `
<label
  spellCheck="true"
/>
`;

exports[`label renders with style={"color":"#000000"} 1`] = `
<label
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`label renders with tabIndex=1 1`] = `
<label
  tabIndex={1}
/>
`;

exports[`label renders with title="TEST" 1`] = `
<label
  title="TEST"
/>
`;

exports[`label renders with translate="no" 1`] = `
<label
  translate="no"
/>
`;

exports[`label renders with translate="yes" 1`] = `
<label
  translate="yes"
/>
`;

exports[`label renders without attributes 1`] = `<label />`;
